include/robot_interfaces/robot_log_reader.hpp

```diff
@@ -52,7 +55,12 @@ class RobotBinaryLogReader
             throw std::runtime_error("Failed to open file " + filename);
         }
 
-        cereal::BinaryInputArchive archive(infile);
+        // wrap the file stream with a gzip decompressor
+        boost::iostreams::filtering_istream infile_compressed;
```

Comment:
I wonder if infile_compressed could be a class attribute. Maybe the decompressor too. To avoid instantiation at each call of the function. (but that may be useless micro-optimization ?)

Reply:
`infile_compressed` contains the file stream to the file passed to this method, so I'm not sure if it would be good to make it a class member.
Typically this method is only called once per instance anyway, so it would probably not make a difference performance-wise.